src/Symfony/Component/Messenger/Bridge/Beanstalkd/Transport/Connection.php

```diff
@@ -187,66 +186,69 @@ private function getFromTube(): ?PheanstalkJob
             }
 
             return $this->client->reserveWithTimeout($this->timeout);
-        } catch (Exception $exception) {
-            throw new TransportException($exception->getMessage(), 0, $exception);
-        }
+        });
     }
 
     public function ack(string $id): void
     {
-        try {
+        $this->withReconnect(function () use ($id) {
             $this->client->useTube($this->tube);
             $this->client->delete(new JobId($id));
-        } catch (Exception $exception) {
-            throw new TransportException($exception->getMessage(), 0, $exception);
-        }
+        });
     }
 
     public function reject(string $id, ?int $priority = null, bool $forceDelete = false): void
     {
-        try {
+        $this->withReconnect(function () use ($id, $priority, $forceDelete) {
             $this->client->useTube($this->tube);
 
             if (!$forceDelete && $this->buryOnReject) {
                 $this->client->bury(new JobId($id), $priority ?? PheanstalkPublisherInterface::DEFAULT_PRIORITY);
             } else {
                 $this->client->delete(new JobId($id));
             }
-        } catch (Exception $exception) {
-            throw new TransportException($exception->getMessage(), 0, $exception);
-        }
+        });
     }
 
     public function keepalive(string $id): void
     {
-        try {
+        $this->withReconnect(function () use ($id) {
             $this->client->useTube($this->tube);
             $this->client->touch(new JobId($id));
-        } catch (Exception $exception) {
-            throw new TransportException($exception->getMessage(), 0, $exception);
-        }
+        });
     }
 
     public function getMessageCount(): int
     {
-        try {
+        return $this->withReconnect(function () {
             $this->client->useTube($this->tube);
             $tubeStats = $this->client->statsTube($this->tube);
-        } catch (Exception $exception) {
-            throw new TransportException($exception->getMessage(), 0, $exception);
-        }
 
-        return $tubeStats->currentJobsReady;
+            return $tubeStats->currentJobsReady;
+        });
     }
 
     public function getMessagePriority(string $id): int
     {
-        try {
+        return $this->withReconnect(function () use ($id) {
             $jobStats = $this->client->statsJob(new JobId($id));
+
+            return $jobStats->priority;
+        });
+    }
+
+    private function withReconnect(callable $command): mixed
+    {
+        try {
+            try {
+                return $command();
+            } catch (ConnectionException) {
+                $this->client->disconnect();
```

Comment:
Not sure what to do about the Psalm errors. Even though the `disconnect()` method wasn't added to the interfaces in v5 (to avoid BC breaks), it was added to the `Pheanstalk` class itself, which is what's used here, so the method always exists. I could use `method_exists()`, but that feels redundant. The method was added to the interfaces in v7, but that version requires PHP 8.3.